app/components/forms/wizard/sponsors-step.js

```diff
@@ -23,39 +23,39 @@ export default Component.extend(FormMixin, {
     };
   },
 
-  sponsors: computed('data.event.sponsors.@each.isDeleted', function() {
-    return this.get('data.event.sponsors').filterBy('isDeleted', false);
+  sponsors: computed('data.sponsors.@each.isDeleted', function() {
+    return this.get('data.sponsors').filterBy('isDeleted', false);
   }),
 
   actions: {
     addSponsor() {
-      this.get('data.event.sponsors').addObject(this.store.createRecord('sponsor'));
+      this.get('data.sponsors').addObject(this.store.createRecord('sponsor'));
     },
     removeSponsor(sponsor) {
-      sponsor.deleteRecord();
+      sponsor.destroyRecord();
```

Comment:
Delete does not ensure that the DELETE request is sent as soon as the user clicks the delete button. With `destroyRecord` as soon as the user clicks on delete button, a DELETE request is sent.

PS. `destroyRecord` works better